serving/src/main/java/feast/serving/service/CachedSpecStorage.java

```diff
@@ -58,14 +59,25 @@ public EntitySpec load(String key) throws Exception {
           }
 
           @Override
-          public Map<String, EntitySpec> loadAll(Iterable<? extends String> keys) throws Exception {
-            return coreService.getEntitySpecs((Iterable<String>) keys);
+          public ListenableFuture<EntitySpec> reload(String key, EntitySpec oldValue)
```

Comment:
I think these can be simplified to:
```
featureSpecLoader = CacheLoader.asyncReloading(
        new CacheLoader<String, FeatureSpec>() {
          @Override
          public FeatureSpec load(String key) throws Exception {
            return coreService.getFeatureSpecs(Collections.singletonList(key)).get(key);
          }
        }, executorService);
```

Reply:
The asyncReloading won't return old value if anything goes wrong during refresh, that's why I used a more verbose implementation.

Reply:
ah neat